rllib/connectors/common/add_observations_from_episodes_to_batch.py

```diff
@@ -95,24 +96,31 @@ def __call__(
         shared_data: Optional[dict] = None,
         **kwargs,
     ) -> Any:
-
-        for sa_episode in self.single_agent_episode_iterator(episodes):
+        # If "obs" already in data, early out.
+        if SampleBatch.OBS in data:
+            return data
+
+        for sa_episode in self.single_agent_episode_iterator(
```

Comment:
Trying (quite successfully) very hard to only use these new ConnectorV2 APIs/utilities from here on:
- single_agent_episode_iterator
- add_batch_item
- add_n_batch_items
- foreach_batch_item_change_in_place